openassessment/templates/openassessmentblock/response/oa_response.html

```diff
@@ -71,7 +71,9 @@ <h4 class="step__title">
                     <ol class="list list--fields response__submission__content">
                         {% for part in saved_response.answer.parts %}
```

Comment:
We now have exactly 3  `submission__answer__part li` parts. To simply achieve our required feature, I would recommend removing loop and hardcoding 3 `li` elements. This way, applying css to last 2 `lis` to show them side by side would be pretty straight forward.

Not a recommended way, but given the urgency, it makes sense to do it for now.

Reply:
Giving 2nd li (code output) `style="display: inline-block"` and 3rd li (expected output) `style="float: right;"`.
And giving `textarea(s)` `rows="10" cols="50"` we can achieve
<img width="1091" alt="Screen Shot 2019-12-26 at 8 05 32 PM" src="https://user-images.githubusercontent.com/22347092/71480489-277db800-281b-11ea-92e9-f057cd15e9df.png">

Reply:
```suggestion
                       <li class="submission__answer__part">  
                                <h5 class="submission__answer__part__text__title">{% trans "The prompt for this section" %}</h5>
                                <article class="submission__answer__part__prompt">
                                    <div class="submission__answer__part__prompt__copy">
                                        {% if prompts_type == 'html' %}
                                            {{ saved_response.answer.parts[0].prompt.description|safe }}
                                        {% else %}
                                            {{ saved_response.answer.parts[0].prompt.description|linebreaks }}
                                        {% endif %}
                                    </div>
                                </article>


                                {% if text_response %}
                                <div class="field field--textarea submission__answer__part__text">
                                    <div class="submission__answer__part__text">
                                        <h5 id="submission__answer__part__text__title__{{ forloop.counter }}__{{ xblock_id }}"
                                            class="submission__answer__part__text__title">
                                            {% trans "Your response (required)" %}
                                        </h5>
                                    </div>
                                    <textarea
                                        id="submission__answer__part__text__1__{{ xblock_id }}"
                                        class="submission__answer__part__text__value"
                                        data-preview="submission__1"
                                        aria-labelledby="submission__answer__part__text__title__1__{{ xblock_id }}"
                                        aria-describedby="submission__answer__tip__{{ xblock_id }}"
                                    >{{ saved_response.answer.parts[0] }}</textarea>
                                </div>
                                {% with "1"|stringformat:"s" as submission_num %}
                                    {% include "openassessmentblock/oa_latex_preview.html" with id="submission__"|add:xblock_id|add:submission_num elem="div" preview_name="submission__"|add:submission_num %}
                                {% endwith %}
                                {% endif %}
                            </li>

                            <!-- code output -->

                            <li class="submission__answer__part" style="display: inline-block";                            >
                                <article class="submission__answer__part__prompt">
                                    <div class="submission__answer__part__prompt__copy">
                                        Output of your code
                                    </div>
                                </article>


                                {% if text_response %}
                                <div class="field field--textarea submission__answer__part__text">
                                    <textarea
                                        id="submission__answer__part__text__2__{{ xblock_id }}"
                                        class="submission__answer__part__text__value"
                                        data-preview="submission__2"
                                        aria-labelledby="submission__answer__part__text__title__2__{{ xblock_id }}"
                                        aria-describedby="submission__answer__tip__{{ xblock_id }}"
                                        rows="10"
                                        cols="50"
                                    >{{ saved_response.answer.parts[1].text }}</textarea>
                                </div>
                                {% with "2"|stringformat:"s" as submission_num %}
                                    {% include "openassessmentblock/oa_latex_preview.html" with id="submission__"|add:xblock_id|add:submission_num elem="div" preview_name="submission__"|add:submission_num %}
                                {% endwith %}
                                {% endif %}
                            </li>


                            <!-- expected output -->

                            <li class="submission__answer__part" style="float: right;">
                                <article class="submission__answer__part__prompt">
                                    <div class="submission__answer__part__prompt__copy">
                                        Expected Output
                                    </div>
                                </article>

                                {% if text_response %}
                                <div class="field field--textarea submission__answer__part__text">
                                    <textarea
                                        id="submission__answer__part__text__3__{{ xblock_id }}"
                                        class="submission__answer__part__text__value"
                                        data-preview="submission__3"
                                        aria-labelledby="submission__answer__part__text__title__3__{{ xblock_id }}"
                                        aria-describedby="submission__answer__tip__{{ xblock_id }}"
                                        maxlength="100000"
                                        rows="10"
                                        cols="50"
                                    >{{ saved_response.answer.parts[2].text }}</textarea>
                                </div>
                                {% with "3"|stringformat:"s" as submission_num %}
                                    {% include "openassessmentblock/oa_latex_preview.html" with id="submission__"|add:xblock_id|add:submission_num elem="div" preview_name="submission__"|add:submission_num %}
                                {% endwith %}
                                {% endif %}
                            </li>
```

Reply:
The above snippet may require a few adjustments.

Reply:
This way we can also add red/green color in borders of 2nd (code output) Textarea based on saved_response.answer.parts[1] value.

Reply:
@noraiz-anwar I will try it. We have 5 `li` not 3(including the 2 for the staff outputs). Since the code(HTML) being used to render the output is nearly the same for both the staff and the learner, there will be some backend changes required as well.

Reply:
you are right. The count is 5.
My suggestion is to not use the loop, so that we can easily style individual `li`  elements. Afterall in our case, we have a fixed number of li to show.
Please let me know if there are any questions.
Thanks.

Reply:
Regarding `code(HTML) being used to render the output is nearly the same for both the staff and the learner`, can we use `request.user.is_staff` to condition rendering in HTML file?

Reply:
From 2nd to onward in prompts, we would need to add checks for `saved_response.answer.parts[index]`e.g add 2nd prompt only if `saved_response.answer.parts[1]` exists and so on.
This will prevent an error that would occur while adding new ora problems.